File: 1.Array/Java/Array.java
class Array {
    public static <T> void show(T arr[], String msg) {
        System.out.println(msg);
        for (T e : arr) {
            System.out.print(e + " ");
        }
        System.out.println();
    }
    public static void main(String args[]) {        
        //array declaration;
        // datatype arr_name[size] = {values...};

        Integer a[] = {1, 2, 3, 4}; // int array
        //int aa[] = {1, 2, 3, 4};
        Float b[] = {1.3f, 3.f, 2.7f, 2f}; // float array
        //float bb[] = {1.3f, 3.f, 2.7f, 2f};
        Double c[] = {2.44, 9.33, 1.2, 89.9}; // double array
        //double cc[] = {2.44, 9.33, 1.2, 89.9};
        Boolean d[] = {true, false, false, true}; // bool arry
        //boolean dd[] = {true, false, false, true};
        Byte e[] = {1, 3, 4, 5}; // byte array
        //byte ee[] = {1, 3, 4, 5};
        String s[] = {"str1", "str2", "str3"}; // string array

        // Two dimentional array
        Integer m[][] = {{1, 2}, {11, 22}, {11, 33}};

        // Three dimentional array
        Integer mm[][][] = {{{1, 2}, {3, 4}}, {{11, 22}, {33,44}}};

        //static array declaration
        // datatype arr_name[size] = {values...}
        int x[] = {1, 2, 4, 5};

        // Dynamic array initialization
        // datatyp arr_name[size] = new dataytpe[size]
        int y[] = new int[4];


        show(a, "Interger array");
        show(b, "Float array");
        show(c, "Double array");
        show(d, "Boolean array");
        show(e, "Byte array");
        show(s, "String array");
    }
}

File: 1.Array/C++/Array.cpp
#include<iostream>
#include<string>
#include<vector>
#include<array>

using namespace std;

template<typename T>
void show(T arr[], int size, string msg) {
    for (int i=0; i<size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main(){
    // declare array
    // datatype array_name[size] = {values...}
    const int size = 4;
    int a[size] = {1, 2, 3, 4}; // Interger Array
    float b[size] = {2.3, 4.3, 5.5}; // Float Array
    double c[size] = {4.5, 63.4, 4.55, 0.0}; // double array
    bool d[size] = {true, false, true}; // boolean array
    
    // string arrays
    string s_0[size] = {"str1", "str2", "str3"};
    const char *s_1[size] = {"str1", "str2", "str3"};
    char s_2[size][10] = {"str7", "str8", "str9"};
    vector<string> s_3 {"str10", "str11", "str12"};
    array<string, size> s_4 {"str13", "str14", "str15"};

    show<int>(a, size, "Integer array");
    show<float>(b, size, "float");
    show<double>(c, size, "double array");
    show<bool>(d, size, "boolean array");
    show<string>(s_0, size, "string_using_string");


    
    return 0;
}